systemtest/src/main/java/io/strimzi/systemtest/utils/RollingUpdateUtils.java

```diff
@@ -0,0 +1,148 @@
+
+public class RollingUpdateUtils {
+    private static final Logger LOGGER = LogManager.getLogger(RollingUpdateUtils.class);
+
+    /**
+     * Method to check that all pods for expected StatefulSet were rolled
+     * @param namespaceName Namespace name
+     * @param selector
+     * @param snapshot Snapshot of pods for StatefulSet before the rolling update
+     * @return true when the pods for StatefulSet are recreated
+     */
+    public static boolean componentHasRolled(String namespaceName, LabelSelector selector, Map<String, String> snapshot) {
+        LOGGER.debug("Existing snapshot: {}", new TreeMap<>(snapshot));
+
+        Map<String, String> currentSnapshot = PodUtils.podSnapshot(namespaceName, selector);
+
+        LOGGER.debug("Current snapshot: {}", new TreeMap<>(currentSnapshot));
+        // rolled when all the pods in snapshot have a different version in map
+
+        currentSnapshot.keySet().retainAll(snapshot.keySet());
+
+        LOGGER.debug("Pods in common: {}", new TreeMap<>(currentSnapshot));
+        for (Map.Entry<String, String> podSnapshot : currentSnapshot.entrySet()) {
+            String currentPodVersion = podSnapshot.getValue();
+            String podName = podSnapshot.getKey();
+            String oldPodVersion = snapshot.get(podName);
+            if (oldPodVersion.equals(currentPodVersion)) {
+                LOGGER.debug("At least {} hasn't rolled", podName);
+                return false;
+            }
+        }
+
+        LOGGER.debug("All pods seem to have rolled");
+        return true;
+    }
+
+    public static boolean componentHasRolled(LabelSelector selector, Map<String, String> snapshot) {
+        return componentHasRolled(kubeClient().getNamespace(), selector, snapshot);
+    }
+
+    /**
+     *  Method to wait when StatefulSet will be recreated after rolling update
+     * @param namespaceName Namespace name
+     * @param selector
+     * @param snapshot Snapshot of pods for StatefulSet before the rolling update
+     * @return The snapshot of the StatefulSet after rolling update with Uid for every pod
+     */
+    public static Map<String, String> waitTillComponentHasRolled(String namespaceName, LabelSelector selector, Map<String, String> snapshot) {
+        String componentName = selector.getMatchLabels().get(Labels.STRIMZI_NAME_LABEL);
+        LOGGER.info("Waiting for component with name: {} rolling update", componentName);
+        TestUtils.waitFor("component with name " + componentName + " rolling update",
+            Constants.WAIT_FOR_ROLLING_UPDATE_INTERVAL, ResourceOperation.timeoutForPodsOperation(snapshot.size()), () -> {
+                try {
+                    return componentHasRolled(namespaceName, selector, snapshot);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                    return false;
+                }
+            });
+
+        LOGGER.info("Component with name: {} has been successfully rolled", componentName);
```

Comment:
Maybe label with some proper formating would be useful as well?

Reply:
Yeah, that makes sense. Thanks :)